substra/sdk/backends/local/backend.py

```diff
@@ -254,7 +254,8 @@ def __format_for_leaderboard(self, testtuple):
         algo = self._db.get(traintuple.algo_type, traintuple.algo.key)
         return {
             'algo': {
-                'hash': algo.key,
+                'key': algo.key,
+                'checksum': algo.checksum,
```

Comment:
algo has no 'checksum' field. This is for the leaderboard, so it returns a subset of the object, and only these fields are printed: https://github.com/SubstraFoundation/substra/blob/master/substra/cli/printers.py#L535

(this function is not tested, whether in debug or normal mode)